File: generated-code/ruby/codecraft/FileReadWrite/model/debug_interface/colored_vertex.rb
require './color'
require './vec2_float'

module Model
module DebugInterface

# Vertex for debug rendering
class ColoredVertex
    # Position in world coordinates (if none, screen position (0, 0) is used)
    attr_accessor :world_pos
    # Additional offset in screen coordinates
    attr_accessor :screen_offset
    # Color to use
    attr_accessor :color

    def initialize(world_pos, screen_offset, color)
        @world_pos = world_pos
        @screen_offset = screen_offset
        @color = color
    end

    # Read ColoredVertex from input stream
    def self.read_from(stream)
        if stream.read_bool()
            world_pos = Vec2Float.read_from(stream)
        else
            world_pos = nil
        end
        screen_offset = Vec2Float.read_from(stream)
        color = Color.read_from(stream)
        ColoredVertex.new(world_pos, screen_offset, color)
    end

    # Write ColoredVertex to output stream
    def write_to(stream)
        if @world_pos.nil?
            stream.write_bool(false)
        else
            stream.write_bool(true)
            @world_pos.write_to(stream)
        end
        @screen_offset.write_to(stream)
        @color.write_to(stream)
    end

    def to_s
        string_result = "ColoredVertex { "
        string_result += "world_pos: "
        if @world_pos.nil?
            string_result += "nil"
        else
            string_result += @world_pos.to_s
        end
        string_result += ", "
        string_result += "screen_offset: "
        string_result += @screen_offset.to_s
        string_result += ", "
        string_result += "color: "
        string_result += @color.to_s
        string_result += " }"
        string_result
    end

    def to_str
        to_s
    end
end

end
end

File: generated-code/python/codecraft/FileReadWrite/model/debug_interface/colored_vertex.py
from color import Color
from stream_wrapper import StreamWrapper
from typing import Optional
from vec2_float import Vec2Float

class ColoredVertex:
    """Vertex for debug rendering"""

    __slots__ = ("world_pos","screen_offset","color",)

    world_pos: Optional[Vec2Float]
    screen_offset: Vec2Float
    color: Color

    def __init__(self, world_pos: Optional[Vec2Float], screen_offset: Vec2Float, color: Color):
        self.world_pos = world_pos
        """Position in world coordinates (if none, screen position (0, 0) is used)"""
        self.screen_offset = screen_offset
        """Additional offset in screen coordinates"""
        self.color = color
        """Color to use"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "ColoredVertex":
        """Read ColoredVertex from input stream
        """
        if stream.read_bool():
            world_pos = Vec2Float.read_from(stream)
        else:
            world_pos = None
        screen_offset = Vec2Float.read_from(stream)
        color = Color.read_from(stream)
        return ColoredVertex(world_pos, screen_offset, color)
    
    def write_to(self, stream: StreamWrapper):
        """Write ColoredVertex to output stream
        """
        if self.world_pos is None:
            stream.write_bool(False)
        else:
            stream.write_bool(True)
            self.world_pos.write_to(stream)
        self.screen_offset.write_to(stream)
        self.color.write_to(stream)
    
    def __repr__(self):
        return "ColoredVertex(" + \
            repr(self.world_pos) + \
            ", " + \
            repr(self.screen_offset) + \
            ", " + \
            repr(self.color) + \
            ")"